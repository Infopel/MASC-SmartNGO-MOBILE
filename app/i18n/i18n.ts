import * as Localization from "expo-localization"
//@ts-ignore
import {I18n} from "i18n-js/dist/require"
import en from "./en"
import pt from "./pt"
export const i18n =  new I18n({en,pt})
i18n.enableFallback = true
i18n.defaultLocale = "en"
// i18n.missingBehavior = "error";

i18n.locale = Localization.locale || "en"

/**
 * Builds up valid keypaths for translations.
 * Update to your default locale of choice if not English.
 */
type DefaultLocale = typeof en
export type TxKeyPath = RecursiveKeyOf<DefaultLocale>

type RecursiveKeyOf<TObj extends Record<string, any>> = {
  [TKey in keyof TObj & string]: TObj[TKey] extends Record<string, any>
  //@ts-ignore  
  ? `${TKey}` | `${TKey}.${RecursiveKeyOf<TObj[TKey]>}`
    : `${TKey}`
}[keyof TObj & string]
