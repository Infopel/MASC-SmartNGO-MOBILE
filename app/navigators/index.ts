export * from "./root-navigator"
export * from "./navigation-utilities"
// export other navigators from here
export * from "./navigators.d"
export * from "./home-bottom-tab/home-bottom-tab-navigator"
export * from "./auth/auth-navigator"
export * from "./primary/primary-navigator"
export * from "./civic-incubator-details-tab/civic-incubator-details-tab-navigator"
export * from "./savings-group-details-tab/savings-group-details-tab-navigator"
