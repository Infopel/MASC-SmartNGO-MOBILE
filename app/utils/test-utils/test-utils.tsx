import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

