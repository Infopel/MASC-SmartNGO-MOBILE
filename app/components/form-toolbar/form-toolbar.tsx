import { Heading } from "components/heading/heading"
import { Icon } from "components/icon/icon"
import { HStack, IconButton, useColorModeValue, useToken, VStack } from "native-base"
import { goBack } from "navigators/navigation-utilities"
import { PrimaryStackHeaderProps } from "navigators/navigators"
import * as React from "react"
import { Platform } from "react-native"
import { Circle } from 'react-native-progress'

export interface FormToolbarProps extends PrimaryStackHeaderProps<'form'> {
  pageInfo?: { stepCount: number; currentStep: number; subTitle: string }
}

/**
 * Describe your component here
 */
export const FormToolbar = function FormToolbar({ pageInfo, route }: FormToolbarProps) {

  const isIOS = Platform.OS === "ios"
  const bg = useColorModeValue("light.50", "dark.100")
  const unfilledProgressColor = useToken('colors', 'success.200')
  const filledProgressColor = useToken('colors', 'success.500')
  const border = useColorModeValue("muted.50", "muted.800")
  return (
    <HStack bg={bg} py={3} px='1' space='1' shadow="1" alignItems={'flex-start'} borderColor={border} borderWidth="1">
      <IconButton p={0} onPress={goBack} testID="back" colorScheme={'text'} icon={<Icon size={'xl'} name='arrow-back' type='material' />} />
      <VStack space='1' flex='1'>

        <Heading fontSize={'xl'} fontWeight={'medium'} noOfLines={1} textAlign={isIOS ? "center" : undefined}
          //@ts-ignore
          tx={`form.${route.params.formId}.title`} />
        {pageInfo && <Heading noOfLines={1} textAlign={isIOS ? "center" : undefined}
          //@ts-ignore
          tx={`form.${route.params.formId}.step-${pageInfo.currentStep}`} fontWeight='light' size='sm' />}
      </VStack>
      {pageInfo && <Circle progress={pageInfo.currentStep / pageInfo.stepCount} showsText formatText={() => pageInfo.currentStep} textStyle={{
        fontSize: 18
      }} color={filledProgressColor} borderWidth={0} unfilledColor={unfilledProgressColor} />}
    </HStack>
  )
}
