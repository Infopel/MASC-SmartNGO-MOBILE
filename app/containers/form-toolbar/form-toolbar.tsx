import { FormToolbar } from "components/form-toolbar/form-toolbar"
import { observer } from "mobx-react-lite"
import { PrimaryStackHeaderProps } from "navigators/navigators"
import * as React from "react"
import { StyleProp, ViewStyle } from "react-native"
import { useFormStore } from "store/form"


export interface FormToolbarProps extends PrimaryStackHeaderProps<"form"> {
  /**
   * An optional style override useful for padding & margin.
   */
  style?: StyleProp<ViewStyle>
}

/**
 * Describe your component here
 */
export const FormToolbarContainer = observer(function (props: FormToolbarProps) {


  const page = useFormStore((state) => state.page())
  const pageInfo = page?.stepCount === 1 || page === undefined ? undefined : {
    stepCount: page.stepCount,
    currentStep: page.index + 1,
    subTitle: page.subTitle
  }
  return (
    <FormToolbar {...props} pageInfo={pageInfo} />
  )
})
