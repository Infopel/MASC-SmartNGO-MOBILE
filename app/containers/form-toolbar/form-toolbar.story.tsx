import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { FormToolbar } from "./form-toolbar.component"

storiesOf("FormToolbar", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <FormToolbar style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
