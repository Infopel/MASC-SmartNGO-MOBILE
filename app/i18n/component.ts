import {TranslateOptions} from 'i18n-js/typings';
import logger from 'utils/logger';
import {TxKeyPath} from './i18n';
import {translate} from './translate';

export interface ITranslatable {
  /**
   * Optional options to pass to i18n. Useful for interpolation
   * as well as explicitly setting locale or translation fallbacks.
   */
  txOptions?: TranslateOptions;
}
export interface ITranslatableText extends ITranslatable {
  /**
   * Text which is looked up via i18n.
   */
  tx?: TxKeyPath;

  /**
   * Optional options to pass to i18n. Useful for interpolation
   * as well as explicitly setting locale or translation fallbacks.
   */
  txOptions?: TranslateOptions;

  /**
   * The text to display if not using `tx` or nested components.
   */

  children?: React.ReactNode;
}

export function translateComponent({
  tx,
  children,
  txOptions,
}: ITranslatableText): React.ReactNode | undefined {
  const i18nText = tx && translate(tx, txOptions);

  if (typeof i18nText !== 'string' && i18nText !== undefined) {
    console.warn('Invalid translated text', {i18nText});
    return children;
  }
  return children || i18nText;
}
