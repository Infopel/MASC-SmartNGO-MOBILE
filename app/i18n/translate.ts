import {TranslateOptions} from 'i18n-js/typings';
import {i18n} from './i18n';
import {TxKeyPath} from './i18n';

/**
 * Translates text.
 *
 * @param key The i18n key.
 */
export function translate(key: TxKeyPath, options?: TranslateOptions) {
  if (!key) throw new Error('Key cannot be nil or empty');
  const trans = i18n.t(key, options);

  if (typeof trans !== 'string')
  throw new Error("The inserted translation key should not map to and object");
  
  return trans;
}
