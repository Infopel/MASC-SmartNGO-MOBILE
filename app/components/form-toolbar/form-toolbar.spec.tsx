import { } from '@react-navigation/native/'
import { translate } from 'i18n/translate'
import { PrimaryStackHeaderProps } from "navigators/navigators"
import * as React from "react"
import { render } from "utils/test-utils/test-utils"
import { FormToolbar } from "./form-toolbar"



describe("<FormToolbar>", () => {
  const navProps: PrimaryStackHeaderProps<'form'> = {
    route: {
      key: "simple key",
      name: "form", params: { formId: 'task-diary' },
    },
    layout: {
      height: 100, width: 100,
    },
    navigation: {

    } as any,
    options: {}, progress: {} as any,
    styleInterpolator: {} as any
  }
  it("render", async function () {
    const {toJSON, queryByTestId} =render( <FormToolbar {...navProps} />)

    expect(queryByTestId('back')).toBeTruthy()
    expect(toJSON()).toMatchSnapshot()
  })
  
  it("Should render form name without page indicator", ()=> {
    const {queryByText} = render(<FormToolbar {...navProps} />)
    expect(queryByText(translate(`form.${navProps.route.params.formId}.title`))).toBeTruthy()
  })
  
  it("Should render page indicator", ()=> {
    const {queryByText} = render(<FormToolbar {...navProps} pageInfo={{stepCount:4, currentStep:2}} />)
    expect(queryByText("2")).toBeTruthy()
    
  })


})
