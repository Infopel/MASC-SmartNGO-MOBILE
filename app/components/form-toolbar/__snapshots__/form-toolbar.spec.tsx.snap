// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormToolbar> render 1`] = `
<View
  dataSet={Object {}}
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#fafaf9",
      "borderColor": "#fafafa",
      "borderWidth": 1,
      "elevation": 2,
      "flexDirection": "row",
      "paddingBottom": 12,
      "paddingLeft": 4,
      "paddingRight": 4,
      "paddingTop": 12,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    colorScheme="text"
    dataSet={Object {}}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 4,
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
    testID="back"
  >
    <Text />
  </View>
  <View
    dataSet={Object {}}
    style={
      Object {
        "width": 4,
      }
    }
  />
  <View
    dataSet={Object {}}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      dataSet={Object {}}
      numberOfLines={1}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#171717",
          "fontFamily": undefined,
          "fontSize": 20,
          "fontStyle": "normal",
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 25,
          "textAlign": "center",
          "textDecorationLine": undefined,
        }
      }
    >
      Task diary
    </Text>
  </View>
</View>
`;
